File: LibraryCSharp/Cipher.cs
using System.Security.Cryptography;
using System.Text;

namespace LibraryCsharp;

public static class Cipher
{
    public static async Task<byte[]> Encrypt(string texto, byte[] key, byte[] iv)
    {
        var aes = Aes.Create();
        aes.Key = key;
        aes.IV = iv;

        // Criação de um criptografador para realizar a transformação de fluxo.
        var encryptor = aes.CreateEncryptor();

        // Criação dos fluxos usados ​​para a criptografia.
        using var memoryStream = new MemoryStream();
        using var cryptoStream = new CryptoStream(memoryStream, encryptor, CryptoStreamMode.Write);
        using (var streamWriter = new StreamWriter(cryptoStream))
        {
            //Gravação de todos os dados no stream.
            await streamWriter.WriteAsync(texto);
        }

        // Retorno dos bytes criptografados do fluxo de memória.
        return memoryStream.ToArray();
    }
}

File: LibraryVisualBasic/Cipher.vb
Imports System.Security.Cryptography
Imports System.IO
Imports LibraryCsharp

Public Class Cipher
    Public Shared Async Function Decrypt(cipherText As Byte(), key As Byte(), iv As Byte()) As Task(Of String)

        Dim aesObj = Aes.Create()
        aesObj.Key = key
        aesObj.IV = iv

        ' Criação de um criptografador para realizar a transformação de fluxo.
        Dim decryptor = aesObj.CreateDecryptor()

        ' Criação dos streams usados ​​para a descriptografia
        Using memoryStreamObj As New MemoryStream(cipherText)
            Using cryptoStream As New CryptoStream(memoryStreamObj, decryptor, CryptoStreamMode.Read)
                Using streamReaderObj As New StreamReader(cryptoStream)
                    'retorno dos bytes descriptografados do fluxo dse descriptografia.
                    Return Await streamReaderObj.ReadToEndAsync()
                End Using
            End Using
        End Using

    End Function
End Class